core/src/main/scala/org/apache/spark/status/AppStatusStore.scala

```diff
@@ -421,6 +420,26 @@ private[spark] class AppStatusStore(
     constructTaskDataList(taskDataWrapperIter)
   }
 
+  def exceptionSummary(stageId: Int, attemptId: Int): Seq[v1.ExceptionSummary] = {
+    val tasks = taskList(stageId, attemptId, Int.MaxValue)
+    tasks.filter(t => t.status.equalsIgnoreCase("failed"))
+      .flatMap(t => t.errorMessage)
+      .flatMap(parseErrorMessage)
+      .groupBy(e => (e.exceptionType, e.message))
+      .map(t => new v1.ExceptionSummary(t._2.head, t._2.length))
+      .toSeq
+      .sortBy(s => (s.count, s.exceptionFailure.exceptionType))(Ordering[(Int, String)].reverse)
+  }
+
+  def parseErrorMessage(errorMessage: String): Option[v1.ExceptionFailure] = {
+    errorMessage.split("\\r?\\n")
+      .find(s => !s.startsWith("\t") && s.contains("Exception:"))
```

Comment:
* Is it possible to have "\r" and "\t" in the message?
* This code assumes the class name contained in the message ends with `Exception` so this can't cover `Error` e.g, OutOfMemoryError.

Reply:
This errorMessage is actually errorMessage with stacktrace. So it includes:
1. First line contains exception type and error message
2. Stacktrace started at `\t` and end with line break.
I'll rewrite this to only include the first line, then split by ":"

Reply:
O.K, I confirmed that stacktrace printed by `printStackTrace` starts with `\t` and each call site is split by `\r\n` on Windows.
But according to the Javadoc, the format, except for the initial line, depends on the implementation.
https://docs.oracle.com/javase/8/docs/api/java/lang/Throwable.html#printStackTrace--

What you want here is the class name and error message, so, can we make this code simpler?

Reply:
Rewrite the logic to handle:
1. exception message is empty
2. Multiple exception types appears will use low level exception type

But still has limitations:
1. As this errorMessage is not exactly `Throwable` errorMessage. It's TaskEndReason.toErrorString, so it could contains something like below:

```
ExecutorLostFailure (executor 389 exited caused by one of the running tasks) Reason: Container from a bad node: container_e1111_1592908128739_15604_01_000741 on host
```
```
FetchFailed(null, shuffleId=516, mapIndex=-1, mapId=-1, reduceId=0, message=
org.apache.spark.shuffle.MetadataFetchFailedException: Missing an output location for shuffle 516
```
2. In the future, we can make TaskEndReason to return exceptionType and message, then we don't parse both from error message.